Simplified/NYPLAccountSignInViewController.m

```diff
@@ -702,26 +704,80 @@ - (void)validateCredentials
   NSMutableURLRequest *const request =
     [NSMutableURLRequest requestWithURL:[NYPLConfiguration loanURL]];
   
-  // Necessary to support longer login times when using usernames.
   request.timeoutInterval = 20.0;
   
-  request.HTTPMethod = @"HEAD";
-  
   self.isCurrentlySigningIn = YES;
   NSURLSessionDataTask *const task =
     [self.session
      dataTaskWithRequest:request
-     completionHandler:^(__attribute__((unused)) NSData *data,
+     completionHandler:^(NSData *data,
                          NSURLResponse *const response,
                          NSError *const error) {
        
        self.isCurrentlySigningIn = NO;
-       // This cast is always valid according to Apple's documentation for NSHTTPURLResponse.
+
        NSInteger const statusCode = ((NSHTTPURLResponse *) response).statusCode;
        
-       // Success.
        if(statusCode == 200) {
+         
+#if defined(FEATURE_DRM_CONNECTOR)        
+
+         NYPLXML *loansXML = [NYPLXML XMLWithData:data];
+         NYPLOPDSFeed *loansFeed = [[NYPLOPDSFeed alloc] initWithXML:loansXML];
+         
+         if (loansFeed.licensor) {
+           [[NYPLAccount sharedAccount] setLicensor:loansFeed.licensor];
+         } else {
+           NYPLLOG(@"Login Failed: No Licensor Token received or parsed from OPDS Loans feed");
+           [self authorizationAttemptDidFinish:NO error:nil];
+           return;
+         }
+         
+         NSMutableArray *licensorItems = [[loansFeed.licensor[@"clientToken"] stringByReplacingOccurrencesOfString:@"\n" withString:@""] componentsSeparatedByString:@"|"].mutableCopy;
+         NSString *tokenPassword = [licensorItems lastObject];
+         [licensorItems removeLastObject];
```

Comment:
It's a good point. I was already mid-way through implementing what you're describing in Swift, and it was making me ask questions around "what should the app allow? or what should be possible? without a valid expected licensor string?". Completing that is still a near-term goal so I'm supporting the compromise for now.